src/lib/cloud/social/provider.ts

```diff
@@ -439,12 +436,79 @@ export class CloudSocialProvider {
       log.warn('cloud contact %1 is not in %2 - cannot remove from storage', host, STORAGE_KEY);
       return Promise.resolve();
     }
+    this.stopMonitoringPresence_(host);
     // Remove host from savedContacts and clients
     delete this.savedContacts_[host];
     delete this.clients_[host];
     // Update storage with this.savedContacts_
     return this.saveContacts_();
   }
+
+  private startMonitoringPresence_(host: string) {
+    if (this.onlinePresenceMonitorIds_[host]) {
+      log.error('unexpected call to startMonitoringPresence_ for ' + host);
+      return;
+    }
+    // Ping server every minute to see if it is online
+    const ping = this.ping_.bind(this, host);
+    const PING_INTERVAL = 60000;
+    this.onlinePresenceMonitorIds_[host] = setInterval(ping, PING_INTERVAL);
+    // Ping server immediately (so we don't have to wait 1 min for 1st result).
+    ping();
+  }
+
+  private stopMonitoringPresence_(host: string) {
+    if (!this.onlinePresenceMonitorIds_[host]) {
+      log.error('unexpected call to stopMonitoringPresence_ for ' + host);
+      return;
+    }
+    clearInterval(this.onlinePresenceMonitorIds_[host]);
+    delete this.onlinePresenceMonitorIds_[host];
+  }
+
+  private ping_(host: string) : Promise<boolean> {
+    var pinger = new Pinger(host, SSH_SERVER_PORT);
+    return pinger.pingOnce().then(() => {
+      return Promise.resolve(true);
```

Comment:
pingOnce() returns a Promise that fulfills if the ping succeeds and rejects if the ping fails.   The purpose of the return true/false is to convert that fulfill/reject into a true/false value so it can be compared to the previous online status.

I did change the `return Promise.resolve(true)` to just `return true` (and same for false)